### PR TITLE
Skip fastparquet timestamp tests when plugin cannot read/write timestamps

### LINKED ISSUE #9776: [BUG] fastparquet compatibility tests fail with data mismatch if TZ is not set and system timezone is not UTC
Running the fastparquet compatibility tests will fail if the TZ environment variable is not set.  I suspect this is because after #9652 we can tell the JVM to use a timezone that could be out-of-sync with what the native code thinks the timezone is.

Here's an example of how I reproduced the errors:
```
$ echo $TZ

$ date
Fri 17 Nov 2023 09:36:39 AM CST
$ SPARK_HOME=/home/jlowe/spark-3.4.1-bin-hadoop3/ integration_tests/run_pyspark_from_build.sh -k test_reading_file_written --pyarrow_test
[...]
FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_by_spark_cpu[Timestamp(not_null)0][DATAGEN_SEED=1700171382, INJECT_OOM] - AssertionError: GPU and CPU timestamp values are different at [0, 'a']
FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_gpu[Timestamp(not_null)0][DATAGEN_SEED=1700171382, INJECT_OOM] - AssertionError: GPU and CPU timestamp values are different at [0, 'a']
```

The tests pass if I export TZ=UTC or fail due to a fallback to the CPU (which is appropriate for the current state of the plugin) if I explicitly set TZ=America/Chicago.

### PR DESCRIPTION
**Pull request by mythrocks**
Fixes #9776.

The tests in `fastparquet_compatibility_test.py` check for compatibility between Apache Spark, the Spark RAPIDS plugin, and fastparquet. In particular:
1. `test_reading_file_written_by_spark_cpu` checks if timestamp columns written with Apache Spark are read similarly with fastparquet and the plugin.
2. `test_reading_file_written_with_gpu` checks if timestamps written with the plugin are read the same on Apache Spark and fastparquet.

If the timezone is not set to `UTC`, and the system timezone isn't `UTC` either, the plugin falls back to CPU for read/write of Parquet timestamp columns. This would cause the above tests not to run: the plugin can neither read nor write timestamps on GPU.

Further, fastparquet seems to interpret timestamps written from Spark as being in `UTC`, regardless of the timezone settings. So on non-UTC timezones, Apache Spark and fastparquet get different results for the same input.

For the two reasons above, it is best to only run the three-way timestamp comparison tests in setups with `UTC` timezone.

This commit skips the timestamp tests described above, when a non-UTC timezone is detected.

### CODE REVIEW

**Comment by mythrocks** [comment]
Build

File: integration_tests/src/main/python/fastparquet_compatibility_test.py [hunk 1]
```diff
@@ -94,6 +94,13 @@ def read_with_fastparquet_or_plugin(spark):
     return read_with_fastparquet_or_plugin
 
 
+def is_timezone_utc():
```
Comment (revans2) [inline integration_tests/src/main/python/fastparquet_compatibility_test.py:97]: There are some APIs that do similar things to this already.

https://github.com/NVIDIA/spark-rapids/blob/b3f1e4744bcee969505cf079a79cd1c308527a97/integration_tests/src/main/python/conftest.py#L83-L87

I think it is better if we use these APIs and then we can improve them if we have a better way to deal with UTC time zone checks.

Comment (mythrocks) [reply integration_tests/src/main/python/fastparquet_compatibility_test.py]: Agreed. Fixed.

**Review by revans2 — APPROVED** [review]

**Comment by mythrocks** [comment]
Build

**Comment by pxLi** [comment]
re-triggered previous github payload empty issue

**Pull request merged**